packages/plugins/graphql-jit/src/index.ts

```diff
@@ -1,18 +1,81 @@
 /* eslint-disable no-console */
 import { Plugin } from '@envelop/types';
-import { GraphQLError } from 'graphql';
-import { compileQuery, isCompiledQuery, CompilerOptions } from 'graphql-jit';
+import { GraphQLError, DocumentNode, ExecutionResult, Source } from 'graphql';
+import { compileQuery, isCompiledQuery, CompilerOptions, CompiledQuery } from 'graphql-jit';
+import lru from 'tiny-lru';
+
+const DEFAULT_MAX = 1000;
+const DEFAULT_TTL = 3600000;
+
+export type JitCacheOptions = {
+  /**
+   * Maximum size of LRU Cache
+   * @default 1000
+   */
+  max?: number;
+  /**
+   * TTL in milliseconds
+   * @default 3600000
+   */
+  ttl?: number;
+};
 
 export const useGraphQlJit = (
   compilerOptions: Partial<CompilerOptions> = {},
-  pluginOptions: Partial<{
-    onError: (r: ReturnType<typeof compileQuery>) => void;
-  }> = {}
+  pluginOptions: {
+    onError?: (r: ReturnType<typeof compileQuery>) => void;
+  } & JitCacheOptions = {}
 ): Plugin => {
+  const max = typeof pluginOptions.max === 'number' ? pluginOptions.max : DEFAULT_MAX;
+  const ttl = typeof pluginOptions.ttl === 'number' ? pluginOptions.ttl : DEFAULT_TTL;
+
+  const DocumentSourceMap = new WeakMap<DocumentNode, string>();
+
+  const jitCache = lru<
```

Comment:
Why do we need to use `ExecutionResult`? Shouldn't it be the result of `compileQuery`?

Reply:
`compileQuery` returns ExecutionResult or CompiledQuery. I think it returns ExecutionResults if the query is not valid or something.

Reply:
I think it's better to type it as `ReturnType<typeof compileQuery>`, the types look way cleaner and are more resilient to changes too, change just added